doc/rpmbuild.8

```diff
@@ -127,44 +127,71 @@ The general form of an rpm build command is
+the stages preceding it), and is one of:
 .TP
 \fB-ba\fR
-Build binary and source packages (after doing the %prep, %build, and
-%install stages).
+Perform a full build - executes up to and including the assembly stage.
+In most cases, this is the option to choose.
 .TP
 \fB-bb\fR
-Build a binary package (after doing the %prep, %build, and %install
-stages).
+Build just the binary packages - executes up to and including the assembly
+stage, but without creating the source package.
 .TP
 \fB-bp\fR
-Executes the "%prep" stage from the spec file. Normally this
-involves unpacking the sources and applying any patches.
+Unpack the sources and apply any patches - executes the %prep stage only.
 .TP
 \fB-bc\fR
-Do the "%build" stage from the spec file (after doing the %prep stage).
+Compile the sources - executes up to and including the %build stage.
 This generally involves the equivalent of a "make".
 .TP
 \fB-bi\fR
-Do the "%install" stage from the spec file (after doing the %prep and
-%build stages).  This generally involves the equivalent of a
-"make install".
+Install the binaries into the build root - executes up to and including the
+%check stage.
+This generally involves the equivalent of a "make install" and "make check".
 .TP
 \fB-bl\fR
-Do a "list check".  The "%files" section from the spec file is
-macro expanded, and checks are made to verify that each file
-exists.
+Do a "list check" - the %files section from the spec file is macro expanded,
+and checks are made to verify that each file exists.
 .TP
 \fB-bs\fR
-Build just the source package.
+Build just the source package - skips straight to the assembly stage, without
+executing any of the preceding stages or creating binary packages.
 .TP
 \fB-br\fR
-Build just the source package - but calculate and include the dynamic build requires.
+Build just the source package, but also parse and include dynamic build
+dependencies - executes up to and including the %generate_buildrequires stage
+and then skips straight to the assembly stage, without creating binary
+packages.
+.IP
+When the %generate_buildrequires stage runs and some of the newly generated
+BuildRequires are not satisfied, \fBrpmbuild\fR creates an intermediate source
+package ending in \fIbuildreqs.nosrc.rpm\fR with all the new BuildRequires, and
+exits with code 11.
+This package can then be used in place of the original source package to
+resolve the missing build dependencies in the usual way.
+.IP
+Multiple levels of dynamic build dependencies may exist in a spec file - a
+particular set of BuildRequires you make available based on a
+\fIbuildreqs.nosrc.rpm\fR package may yield another set of BuildRequires next
+time a build is performed with the same source package.
+The easiest way to ensure that all dynamic build dependencies are satisfied is
+to run this command (followed by new dependency resolution) repeatedly until it
+no longer exits with code 11.
```

Comment:
```suggestion
to run this command (followed by installation of the generated dependencies) repeatedly
until no new dependencies are generated.
```

Probably something like this?

Reply:
Yeah, "new dependency resolution" sounds a bit awkward and isn't exactly clear. Will fix.
As for the "until" clause, I wonder if rpmbuild can exit with 0 and still generate new deps? From what I can see in the code, this shouldn't ever happen. I know that Mock actually does check for new deps (instead of the return code), but that's because they have to use `--nodeps` (due to unrelated reasons) which always returns 11.

Reply:
OK, thinking about it more, a situation could arise where a missing dep can't be resolved or installed (with `dnf builddep`) for whatever reason, in which case saying that "repeat the whole `rpmbuild -br` + `dnf builddep` process while exit code = 11" would make this into an endless loop :)

Reply:
New revision force-pushed.